From e7392bbacb26dd18f54854165cd39fda871d8a60 Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Fri, 3 Apr 2020 20:29:52 +0800
Subject: [PATCH] mxc: fix mxc/vpu-decoder-b0 and mxc/vpu-encoder-b0 build
 error

fix mxc/vpu-decoder-b0 and mxc/vpu-encoder-b0 build error.
warning: 'strncpy' specified bound 32 equals destination size
[-Wstringop-truncation] appears when compiling

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/vpu_b0.c         | 2 +-
 drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
index bb11695a8eda..1a2d0456539c 100644
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -3376,7 +3376,7 @@ static int create_vpu_video_device(struct vpu_dev *dev)
 		vpu_dbg(LVL_ERR, "video device alloc for decoder fail\n");
 		return -ENOMEM;
 	}
-	strncpy(dev->pvpu_decoder_dev->name, v4l2_videodevice_decoder.name, sizeof(v4l2_videodevice_decoder.name));
+	strscpy(dev->pvpu_decoder_dev->name, v4l2_videodevice_decoder.name, sizeof(v4l2_videodevice_decoder.name));
 	dev->pvpu_decoder_dev->fops = v4l2_videodevice_decoder.fops;
 	dev->pvpu_decoder_dev->ioctl_ops = v4l2_videodevice_decoder.ioctl_ops;
 	dev->pvpu_decoder_dev->release = video_device_release;
diff --git a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
index 5a6b8c255ffb..f824165165d0 100644
--- a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
+++ b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
@@ -4590,7 +4590,7 @@ static int create_vpu_video_device(struct vpu_dev *dev)
 		return -ENOMEM;
 	}
 
-	strncpy(dev->pvpu_encoder_dev->name,
+	strscpy(dev->pvpu_encoder_dev->name,
 		v4l2_videodevice_encoder.name,
 		sizeof(v4l2_videodevice_encoder.name));
 	dev->pvpu_encoder_dev->fops = v4l2_videodevice_encoder.fops;
-- 
2.17.1

